tree.go

```diff
@@ -1492,21 +1521,22 @@ func (n *LeafNode) GetProofItems(keys keylist, _ NodeResolverFn) (*ProofElements
 			}
 		}
 
-		// Proof of absence: case of a missing suffix tree.
-		//
-		// The suffix tree for this value is missing, i.e. all
-		// values in the extension-and-suffix tree are grouped
-		// in the other suffix tree (e.g. C2 if we are looking
-		// at C1).
-		if count == 0 {
-			// TODO(gballet) maintain a count variable at LeafNode level
-			// so that we know not to build the polynomials in this case,
-			// as all the information is available before fillSuffixTreePoly
-			// has to be called, save the count.
-			esses = append(esses, extStatusAbsentEmpty|(n.depth<<3))
-			pe.Vals = append(pe.Vals, nil)
-			continue
-		}
+		_ = count
+		// // Proof of absence: case of a missing suffix tree.
+		// //
+		// // The suffix tree for this value is missing, i.e. all
+		// // values in the extension-and-suffix tree are grouped
+		// // in the other suffix tree (e.g. C2 if we are looking
+		// // at C1).
+		// if count == 0 {
+		// 	// TODO(gballet) maintain a count variable at LeafNode level
+		// 	// so that we know not to build the polynomials in this case,
+		// 	// as all the information is available before fillSuffixTreePoly
+		// 	// has to be called, save the count.
+		// 	esses = append(esses, extStatusAbsentEmpty|(n.depth<<3))
+		// 	pe.Vals = append(pe.Vals, nil)
+		// 	continue
+		// }
```

Comment:
I'm not sure this was the cause of issues, but definitely I think we need to chat about this.

This was adding a `extStatusAbsentEmpty`, and that's very tricky. 
I totally understand the intention that if we are proving **a single key** which lies in an empty half of the leaf node, then we could use `extStatusAbsentEmpty` since the verifier could assume there's no LeafNode here and get a `nil`.

Note that this is a trick and not representative of the real tree. The real tree **has** a LeafNode, but for this particular situation we can "simulate" that doesn't.

Let's assume for a moment we're OK with that.

Still, the code here isn't doing things correctly. We can add this `extStatusAbsentEmpty` if there's **a single key** proven in this way. As in, if for a key we did this addition of `extStatusAbsentEmpty`, and then the next key proves the presence in a non-empty half, then we should definitely removed that `extStatusAbsentEmpty` thing... if not, we'd have more than one extension status for keys of this leaf.

_Maybe_ it's something that we can try to optimize. But it's not correct to simply add this empty extension status. We need to have some internal state to track that we did that, and if anything else happens in a next key (i.e: proof of absence other, or presence) then we need to revert that addition.

We can keep chatting in Matrix about this. I just wanted to leave some comment since I have this fresh in my mind now.

Reply:
> Still, the code here isn't doing things correctly. We can add this extStatusAbsentEmpty if there's a single key proven in this way. As in, if for a key we did this addition of extStatusAbsentEmpty, and then the next key proves the presence in a non-empty half, then we should definitely removed that extStatusAbsentEmpty thing... if not, we'd have more than one extension status for keys of this leaf.

Yes, the extra empty was being removed if a proof of presence was found. That was line 1474, when setting `esses = nil`.

It's ok, let's keep things simple and have one extension status per stem.